src/IR.h

```diff
@@ -718,6 +717,25 @@ struct Prefetch : public StmtNode<Prefetch> {
     static const IRNodeType _type_info = IRNodeType::Prefetch;
 };
 
+struct AddressOf : public ExprNode<AddressOf> {
```

Comment:
Needs a comment explaining what the meaning is. Something like "Equivalent to a Load node, but represents the address that would be loaded from rather than the value".